app/src/main/java/com/example/unifood/recyclerview/RestaurantRecyclerView.kt

```diff
@@ -39,9 +48,31 @@ class RestaurantListAdapter(private val items: List<Restaurant>) :
             itemView.findViewById<TextView>(R.id.textState1).text =
                 if (item.isOpen) "Open" else "Closed"
             itemView.findViewById<TextView>(R.id.textPrice).text = item.avgPrice.formatAvgPrice()
-            itemView.findViewById<TextView>(R.id.textDistance).text =
-                "${item.distance.onlyTwoDecimals()} km away"
+
+            // Corrigiendo el tipo de dato para el formato de distancia
+            val distanceText = itemView.context.getString(R.string.restaurant_distance, item.distance.onlyTwoDecimals().toDouble())
```

Comment:
You convert a Double into a String with `.onlyTwoDecimals()` and to Double again, this is causing an exception.

`java.lang.NumberFormatException:` For input string: "0,4"`